lib/databases/postgresql

```diff
@@ -80,16 +87,22 @@ function configure_database_postgresql {
     sudo sed -i "/max_connections/s/.*/max_connections = $MAX_DB_CONNECTIONS/" $pg_conf
     # Do password auth from all IPv4 clients
     sudo sed -i "/^host/s/all\s\+127.0.0.1\/32\s\+ident/$DATABASE_USER\t0.0.0.0\/0\tpassword/" $pg_hba
+    sudo sed -i "/^host/s/all\s\+127.0.0.1\/32\s\+reject/$DATABASE_USER\t0.0.0.0\/0\tpassword/" $pg_hba
```

Comment:
I am unsure why these reject lines are needed?  If I understand how pg_hba.conf works they will never be reached since the selection criteria match exactly the ident line above them.

Reply:
On Clearlinux, the origin pg_hba.conf looks like this:
# "local" is for Unix domain socket connections only
local   all             all                                     reject
# IPv4 local connections:
host    all             all             127.0.0.1/32            reject
# IPv6 local connections:
host    all             all             ::1/128                 reject

If the image can fix this, we dont need these code.

Reply:
Does that change work?  This is meant to control who can connect to psql, and for multi-node jobs it needs to be more than localhost (the normal default).  I think what you need is to also rewrite those reject lines to ident lines.  I am not a psql expert, but what I see on my Ubuntu devstack is:

local   all             all                                     peer
host    all             all             127.0.0.1/32            md5
host    all             all             ::1/128                 md5

I don't think you want to reject connections...

Reply:
I mean that's the original config file.
What I did in the pr is to change the "reject" to "password".
But I still need one more line to modify "local all all reject" to "local all all ident". Then everything  works fine.

But, I think the original config file is not right, it should be the same as other systems, so that we dont have to change this code.

Reply:
Ah, OK, I didn't follow the regex properly, of course that's right given what you have to start with.  Thanks for being patient with me. :)

Reply:
That's ok, the cmd line is toooo long to understand.
Shall we keep this change as a temp patch? So that if the clear team modified the original conf file, we dont have to merge this one.
I would seperate the installing code to another patch.